### PR TITLE
Fixed #35091 -- Allowed GeoIP2 querying using IPv4Address/IPv6Address.

### PR DESCRIPTION
**Pull request by ngnpope**
ticket-35091.

### CODE REVIEW

File: django/contrib/gis/geoip2.py [hunk 1]
```diff
@@ -173,9 +174,10 @@ def __repr__(self):
     def _check_query(self, query, city=False, city_or_country=False):
         "Check the query and database availability."
         # Making sure a string was passed in for the query.
-        if not isinstance(query, str):
+        if not isinstance(query, (str, ipaddress.IPv4Address, ipaddress.IPv6Address)):
```
Comment (felixxm) [inline django/contrib/gis/geoip2.py:177]: Update the comment above.

File: docs/releases/5.1.txt [hunk 2]
```diff
@@ -59,6 +59,9 @@ Minor features
 * :class:`~django.contrib.gis.db.models.Collect` is now supported on MySQL
   8.0.24+.
 
+* :class:`~django.contrib.gis.geoip2.GeoIP2` now supports querying using
```
Comment (felixxm) [inline docs/releases/5.1.txt:62]:
```suggestion
* :class:`~django.contrib.gis.geoip2.GeoIP2` now allows querying using
```

File: django/contrib/gis/geoip2.py (hunk 1 above)
Comment (ngnpope) [reply django/contrib/gis/geoip2.py:177]: I think we can drop the comment. It doesn't add value - the `isinstance(...)` is quite clear enough.

**Comment by felixxm** [comment]
@ngnpope Thanks :+1:

**Pull request merged**